### PR TITLE
Code factor refactors

### PR DESCRIPTION
**Pull request by sdimitris**
Some cleanups to make code factor happier :)
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch
- [ ] When resolving a specific issue, it's referenced in the PR's description  (e.g. `fix #xxx`, where "xxx" is the issue number)

### CODE REVIEW

File: modules/authentication/src/handlers/local.ts [hunk 1]
```diff
@@ -285,9 +285,11 @@ export class LocalHandlers implements IAuthenticationStrategy {
   }
 
   async register(call: ParsedRouterRequest): Promise<UnparsedRouterResponse> {
-    let { email, password } = call.request.params;
+    let email = call.request.params.email;
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:288]: Make this a const:
`const email = call.request.params.email.toLowerCase();`

File: modules/authentication/src/utils/auth.ts [hunk 2]
```diff
@@ -7,18 +7,68 @@ import ConduitGrpcSdk, {
...
 import { status } from '@grpc/grpc-js';
 import { Config } from '../config';
+import { v4 as uuid } from 'uuid';
 
 export namespace AuthUtils {
   export function randomToken(size = 64) {
     return crypto.randomBytes(size).toString('base64');
   }
 
+  export async function tokenRecreation(
```
Comment (kon14) [inline modules/authentication/src/utils/auth.ts:24]: Rename this to `createToken` for readability.

File: modules/authentication/src/handlers/local.ts [hunk 3]
```diff
@@ -933,7 +891,7 @@ export class LocalHandlers implements IAuthenticationStrategy {
     const { method, phoneNumber } = call.request.params;
     const context = call.request.context;
 
-    if (isNil(context) || isNil(context.user)) {
+    if (isNil(context) || isEmpty(context.user)) {
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:894]: Revert this as `user` field is not an array-like.

File: modules/authentication/src/handlers/local.ts [hunk 4]
```diff
@@ -864,7 +822,7 @@ export class LocalHandlers implements IAuthenticationStrategy {
     });
     if (!isNil(verificationToken)) {
       const diffInMilliSec = Math.abs(
-        new Date(verificationToken.createdAt).getTime() - Date.now(),
+        new Date(verificationToken!.createdAt).getTime() - Date.now(),
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:825]: This should be redundant (?)

File: modules/authentication/src/handlers/local.ts [hunk 5]
```diff
@@ -650,56 +634,31 @@ export class LocalHandlers implements IAuthenticationStrategy {
...
+        TokenType.CHANGE_EMAIL_TOKEN,
+      ).catch(e => {
+        ConduitGrpcSdk.Logger.error(e);
       });
       if (this.sendVerificationEmail) {
         const serverConfig = await this.grpcSdk.config.get('router');
         const url = serverConfig.hostUrl;
         const result = { verificationToken, hostUrl: url };
-        const link = `${result.hostUrl}/hook/authentication/verify-change-email/${result.verificationToken.token}`;
+        const link = `${result.hostUrl}/hook/authentication/verify-change-email/${result.verificationToken?.token}`;
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:661]: This should be redundant (?)

File: modules/authentication/src/handlers/local.ts [hunk 6]
```diff
@@ -512,7 +515,7 @@ export class LocalHandlers implements IAuthenticationStrategy {
       '+hashedPassword',
     );
     if (isNil(user)) throw new GrpcError(status.NOT_FOUND, 'User not found');
-    if (isNil(user.hashedPassword))
+    if (isNil(user?.hashedPassword))
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:518]: This is redundant. We already verify user exists in the above line. If an assertion is required, use `!`.

**Review by kon14 — CHANGES_REQUESTED** [review]
tbh I'd also remove the `invalidAddress` variables as the inline call was pretty clear.
Thanks CodeFactor.

**Review by kkopanidis — APPROVED** [review]

**Review by kon14 — APPROVED** [review]

**Review by ChrisPdgn — APPROVED** [review]

**Pull request merged**